Authentication/SignUp/View/SignupController.swift

```diff
@@ -0,0 +1,230 @@
+//
+//  SignupController.swift
+//  Authentication
+//
+//  Created by Daniela Riesgo on 3/7/16.
+//  Copyright © 2016 Wolox. All rights reserved.
+//
+
+import Foundation
+import ReactiveCocoa
+
+public final class SignupController: UIViewController {
+    
+    private let _viewModel: SignupViewModelType
+    private let _signupViewFactory: () -> SignupViewType
+    private let _delegate: SignupControllerDelegate
+    
+    public lazy var signupView: SignupViewType = self._signupViewFactory()
+    
+    private lazy var _notificationCenter: NSNotificationCenter = NSNotificationCenter.defaultCenter()
+    private var _disposable = CompositeDisposable()
+    private let _keyboardDisplayed = MutableProperty(false)
+    private let _activeTextField = MutableProperty<UITextField?>(.None)
+
+    
+    init(viewModel: SignupViewModelType, signupViewFactory: () -> SignupViewType, delegate: SignupControllerDelegate = DefaultSignupControllerDelegate()) {
```

Comment:
I think we should specify why this is internal or/and to label it `internal` to avoid confusion.